icu4c/source/tools/ctestfw/unicode/testtype.h

```diff
@@ -39,3 +39,9 @@
     #define T_CTEST_EXPORT_API T_CTEST_IMPORT
 #endif
 
+#ifndef TRUE
```

Comment:
Why?  Weren't we going to use a C-specific header from the standard library (I forgot which one)?

Reply:
I am not offering to change all of the implementation/samples/tools/test code in this ticket+PR. Therefore, the macros continue to be defined when compiling the ICU libraries themselves, and I added equivalent definitions into a few strategic places so that I need not change another few hundred files.

Only in simple cases that don't benefit from the shared toolutil or test header files did I change the call sites to use false/true.